operations/jsonnet/microservices/compactor.libsonnet

```diff
@@ -28,7 +29,8 @@
     ]) +
     $.util.withResources($._config.compactor.resources) +
     $.util.readinessProbe +
-    (if $._config.variables_expansion then container.withArgsMixin(['--config.expand-env=true']) else {}),
+    (if $._config.variables_expansion then container.withArgsMixin(['--config.expand-env=true']) else {}) +
+    container.withEnvMixin([envVar.new('GOMEMLIMIT', $._config.compactor.resources.limits.memory + 'B')]),
```

Comment:
if someone were to set `_config.compactor.resources.limits.memory` to null i think this would break. i think an if statement like the above should be able to check before adding the env var.